R/prior.R

```diff
@@ -90,12 +90,20 @@ epidist_family_prior.default <- function(family, formula, ...) {
 epidist_family_prior.lognormal <- function(family, formula, ...) {
   prior <- brms::prior("normal(1, 1)", class = "Intercept")
   if ("sigma" %in% names(formula$pforms)) {
-    prior <- prior +
-      brms::prior("normal(-0.7, 0.4)", class = "Intercept", dpar = "sigma")
+    # Case with a model on sigma
+    sigma_prior <- brms::prior(
+      "normal(-0.7, 0.4)", class = "Intercept", dpar = "sigma"
+    )
+  } else if ("sigma" %in% names(formula$pfix)) {
+    # Case with sigma fixed to a constant
+    sigma_prior <- NULL
   } else {
-    prior <- prior +
-      brms::prior("lognormal(-0.7, 0.4)", class = "sigma", lb = 0, ub = "NA")
+    # Case with no model on sigma
+    sigma_prior <- brms::prior(
```

Comment:
not for now but it seems like the solution above could be applied here quite easily if you also knew the link (at least in the log case).

Reply:
Solution above as in putting things into `epidist_prior`?

Reply:
yes and trying to automate vs on a case by case basis

Reply:
Yeah I agree about trying to automate this as we add more priors

Reply:
https://github.com/epinowcast/epidist/issues/305

Reply:
I think this needs its own issue as its a slightly different problem to the no model question (though related) and may not need to be dealt with in its own PR.

Reply:
https://github.com/epinowcast/epidist/issues/307